evm/src/cpu/stack.rs

```diff
@@ -270,10 +299,22 @@ pub(crate) fn eval_packed<P: PackedField>(
     nv: &CpuColumnsView<P>,
     yield_constr: &mut ConstraintConsumer<P>,
 ) {
-    for (op, stack_behavior) in izip!(lv.op.into_iter(), STACK_BEHAVIORS.into_iter()) {
+    for (op, stack_behavior, might_overflow) in izip!(
+        lv.op.into_iter(),
+        STACK_BEHAVIORS.into_iter(),
```

Comment:
I must be missing something. The `izip` macro will terminate when one of the sub-iterators return `None`, so this `for`-loop will stop when we get to `eq_iszero`? Is that the intended behaviour here? Or am I misunderstanding this completely?

Reply:
Ok, ignore me. I get it.

I wonder though: can we refactor the `STACK_BEHAVIOR` to not be a list of `Option<StackBehavior>` and add the info in `MIGHT_OVERFLOW` to the `StackBehavior` type instead? It seems like most ops have some kind of "special flair", so perhaps it is wise to collect all that metadata into one type?

Reply:
That's a good idea, and I think I even went for this approach the first time I implemented it. I think I changed my mind because there are other similar structs like `KEEPS_CONTEXT` in `contextops.rs` and `SIMPLE_OPCODES` in `gas.rs`.
Do you think it would make sense to have a single struct called `INSTRUCTION_BEHAVIOR` or something with all these combined?

Reply:
Intuitively yes, I think so. Given this is a hot struct (right?), it should be as small as possible so we need to be careful that the layout is cache friendly, but I bet that traversing several smaller lists/structs is costlier. Hard numbers would be good ofc.

Reply:
This is noted, but this requires enough changes to tackle it in another PR in my opinion.